obs2ioda-v2/src/ncio_mod.f90

```diff
@@ -176,90 +207,115 @@ subroutine write_obs (filedate, write_opt, outdir, itim)
          ncname = trim(name_var_info(i))
-         igrp = ufo_vars_getindex(name_ncgrp, 'MetaData')
-         idim = ufo_vars_getindex(name_ncdim, dim_var_info(1,i))
-         dim1 = ncid_ncdim(idim)
          if ( ufo_vars_getindex(name_ncdim, dim_var_info(2,i)) > 0 ) then
+            idim = ufo_vars_getindex(name_ncdim, dim_var_info(1,i))
+            dim1 = ncid_ncdim(idim)
+            dim1_name = get_dim_name(ncid_ncdim(dim1), nchans_nvars_flag)
             idim = ufo_vars_getindex(name_ncdim, dim_var_info(2,i))
             dim2 = ncid_ncdim(idim)
-            call def_netcdf_var(ncid_ncgrp(igrp),ncname,(/dim1,dim2/),type_var_info(i))
+            dim2_name = get_dim_name(ncid_ncdim(dim2), nchans_nvars_flag)
+            status = netcdfAddVar(netcdfID, ncname, NF90_CHAR, 2, &
+               [dim2_name, dim1_name], "MetaData")
          else
-            if ( ncname == 'dateTime' ) then
-               call def_netcdf_var(ncid_ncgrp(igrp),ncname,(/dim1/),type_var_info(i), &
-                  'units', 'seconds since 1970-01-01T00:00:00Z')
+            idim = ufo_vars_getindex(name_ncdim, dim_var_info(1,i))
+            dim1 = ncid_ncdim(idim)
+            if (ncname == 'dateTime') then
+               dim1_name = get_dim_name(ncid_ncdim(dim1), nchans_nvars_flag)
+               status = netcdfAddVar(netcdfID, ncname, type_var_info(i), 1, &
+                  [dim1_name], "MetaData")
+               status = netcdfPutAtt(netcdfID, "units", "seconds since 1970-01-01T00:00:00Z", varName = trim(ncname), &
+                  groupName = "MetaData")
             else
-               call def_netcdf_var(ncid_ncgrp(igrp),ncname,(/dim1/),type_var_info(i))
+               dim1_name = get_dim_name(ncid_ncdim(dim1), nchans_nvars_flag)
+               status = netcdfAddVar(netcdfID, ncname, type_var_info(i), 1, &
+                  [dim1_name], "MetaData")
+               if (type_var_info(i) == NF90_INT) then
+                  status = netcdfSetFill(netcdfID, ncname, 1, -999, "MetaData")
+               else if (type_var_info(i) == NF90_FLOAT) then
+                  status = netcdfSetFill(netcdfID, ncname, 1, -999.0, "MetaData")
+               end if
             end if
          end if
       end do var_info_def_loop ! nvar_info
 
       if ( write_opt == write_nc_radiance .or. write_opt == write_nc_radiance_geo ) then
          do i = 1, nsen_info
             ncname = trim(name_sen_info(i))
-            igrp = ufo_vars_getindex(name_ncgrp, 'MetaData')
             idim = ufo_vars_getindex(name_ncdim, dim_sen_info(1,i))
             dim1 = ncid_ncdim(idim)
             if ( ufo_vars_getindex(name_ncdim, dim_sen_info(2,i)) > 0 ) then
                idim = ufo_vars_getindex(name_ncdim, dim_sen_info(2,i))
                dim2 = ncid_ncdim(idim)
-               call def_netcdf_var(ncid_ncgrp(igrp),ncname,(/dim1,dim2/),type_sen_info(i))
+               dim1_name = get_dim_name(ncid_ncdim(dim1), nchans_nvars_flag)
```

Comment:
Can this line be factored out of the if-else block and moved directly below line 245?